frontend/public/components/modals/configure-count-modal.tsx

```diff
@@ -62,93 +83,117 @@ export const ConfigureCountModal = (props: ConfigureCountModalProps) => {
-        titleKey: 'public~Edit Pod count',
-        labelKey: props.resourceKind.labelPluralKey,
-        // t('public~{{resourceKinds}} maintain the desired number of healthy pods.', {resourceKind: props.resourceKind.labelPlural})
-        messageKey: 'public~{{resourceKinds}} maintain the desired number of healthy pods.',
-        messageVariables: { resourceKinds: props.resourceKind.labelPlural },
-        path: '/spec/replicas',
-        // t('public~Save')
-        buttonTextKey: 'public~Save',
-        opts: { path: 'scale' },
-      },
-      props,
-    ),
+  const mergedProps = _.assign(
+    {},
+    {
+      defaultValue: 0,
+      titleKey: 'public~Edit Pod count',
+      labelKey: props.resourceKind.labelPluralKey,
+      messageKey: 'public~{{resourceKinds}} maintain the desired number of healthy pods.',
+      messageVariables: { resourceKinds: props.resourceKind.labelPlural },
+      path: '/spec/replicas',
+      buttonTextKey: 'public~Save',
+      opts: { path: 'scale' },
+    },
+    props,
   );
+
+  // Return the props for the useOverlay pattern
+  return mergedProps;
 };
 
 export const configureJobParallelismModal = (props) => {
-  return configureCountModal(
-    _.defaults(
-      {},
-      {
-        defaultValue: 1,
-        // t('public~Edit parallelism')
-        titleKey: 'public~Edit parallelism',
-        // t('public~{{resourceKinds}} create one or more pods and ensure that a specified number of them successfully terminate. When the specified number of completions is successfully reached, the job is complete.', {resourceKind: props.resourceKind.labelPlural})
-        messageKey:
-          'public~{{resourceKinds}} create one or more pods and ensure that a specified number of them successfully terminate. When the specified number of completions is successfully reached, the job is complete.',
-        messageVariables: { resourceKinds: props.resourceKind.labelPlural },
-        path: '/spec/parallelism',
-        // t('public~Save')
-        buttonTextKey: 'public~Save',
-      },
-      props,
-    ),
+  const mergedProps = _.assign(
+    {},
+    {
+      defaultValue: 1,
+      titleKey: 'public~Edit parallelism',
+      messageKey:
+        'public~{{resourceKinds}} create one or more pods and ensure that a specified number of them successfully terminate. When the specified number of completions is successfully reached, the job is complete.',
+      messageVariables: { resourceKinds: props.resourceKind.labelPlural },
+      path: '/spec/parallelism',
+      buttonTextKey: 'public~Save',
+    },
+    props,
   );
+
+  // Return the props for the useOverlay pattern
+  return mergedProps;
 };
 
 export type ConfigureCountModalProps = {
   message?: string;
-  messageKey: string;
-  messageVariables: { [key: string]: any };
+  messageKey?: string;
+  messageVariables?: Record<string, string>;
   buttonText?: string;
   buttonTextKey?: string;
   buttonTextVariables?: { [key: string]: any };
```

Comment:
Pretty sure this can be retyped to `Record<string, string>` too.
```suggestion
  buttonTextVariables?: Record<string, string>;
```